api/v1beta1/cluster_webhook.go

```diff
@@ -197,6 +197,18 @@ func (c *Cluster) validateTopology(old *Cluster) field.ErrorList {
 			)
 		}
 	default: // On update
+		if old.Spec.Topology == nil || old.Spec.Topology.Class == "" {
+			allErrs = append(
+				allErrs,
+				field.Forbidden(
+					field.NewPath("spec", "topology", "class"),
+					"class cannot be set on an existing Cluster",
```

Comment:
```suggestion
					"cannot be set on an existing Cluster",
```
nit for consistency (we didn't mention the field name in the cases above)

I wonder if we should call it an `existing unmanaged Cluster` or something

Reply:
This is in a similar state on main. Should I update there too? i was thinking of adding the unit test anyway.

Reply:
On the other hand we have "class cannot be changed" directly below this. I think we should merge this one as is, mirroring 1.1, and maybe have another look at the messages we use on main to provide a better UX going forward.

Reply:
Good point please ignore the class nit :)

Oh, I didn't take a look at the base branch. I think we usually prefer fixing the issue on main first and then backporting (if necessary with equivalent commits, if the same doesn't work).

On which branches do we have the panic? Or is it only on release-1.0?

Reply:
Only 1.0. It's fixed in 1.1 and 0.4 doesn't have these sorts of checks.